lib/include/tree_sitter/api.h

```diff
@@ -1166,6 +1166,26 @@ uint32_t ts_language_symbol_count(const TSLanguage *self);
 */
 uint32_t ts_language_state_count(const TSLanguage *self);
 
+/**
+ * Get the number of supertype symbols in the language.
+*/
+uint32_t ts_language_supertype_count(const TSLanguage *self);
+
+/**
+ * Get a list of all supertype symbols for the language.
+*/
+const TSSymbol *ts_language_supertypes(const TSLanguage *self);
+
+/**
+ * Get a list of all subtype symbol names for a given supertype symbol. Returns
+ * the size of said list.
+*/
+uint16_t ts_language_supertype_map(
+  const TSLanguage *self,
+  TSSymbol supertype,
+  const char *const **start
```

Comment:
I don't think we should be writing C-style strings, as this function is giving us the subtype symbol names, of which we can't really do much with. It would make a lot more sense imo, to store the supertype info in the parser as a map of symbol ids to an array of symbol ids, which is a lot better for the parser binary size and memory usage (esp in wasm). With the symbol ids, they are a lot more useful to us and we can always fetch the symbol names with this array of ids.